pyladie_oop/clovek.py

```diff
@@ -0,0 +1,46 @@
+class Clovek:
+    def __init__(self, name) -> None:
+
+        self.name = name
+        self.focus = 3  # minimum 0
+        self.max_focus = 3
+        self.energy = 5  # minimum 0
+        self.max_energy = 5
+
+    def eat(self):
+        # zvyšuje energy, připadně focus
+        if self.energy >= 5 and self.focus >= 3:
```

Comment:
Tu k tomu mám dve veci:
1. Tento if sa splní len za predpokladu, že platia obe porovnania. Ak teda budeme v situácii, kedy má plnú energiu ale nie focus, tak sa dostaneme do else kde energiu navýšiš a potom ju pomocou self.max_control() zasa vrátiš na 5. To je za mňa kopa zbytočnej práce a nedáva zmysel. Za mňa by dávalo zmysel vytvoriť si metódu, ktorej predáš argumenty `self.energy`, `self.max_energy`, minimálnu hodnotu a `1` v prípade napr. riadku 15 (môžeš to teda použiť aj na znižovanie hodnoty). Táto metóda zaručí, že sa daná vec navýši/zníži o určitú hodnotu ale len do max./min. povolenej hodnoty. Môžeš to volať s danými argumentami vždy keď chceš znižovať/zvyšovať hodnotu a nemusíš riešiť nejaké dodatočné úpravy, kontroly, ify a pod.
```
def change_within_limits(self, subject, value, max_v, min_v = 0):
    return max(min(max_v, subject + value), min_v)
```
Minimálnu hodnotu si môžeš nastaviť ako konštantu na 0 a stačí ti pri volaní metódy predať argumenty subject, value, max_v.

2. Keď už máš maximum definované v `self.max_energy`, porovnávala by som to s ním aby si zaistila, že ak by sa napr. v nejakej odvodenej triede zmenil argument `self.max_energy`, tak ti metóda `eat` bude stále fungovať správne. To isté platí pre focus.